backend/telemetry/src/main/java/org/sonarsource/sonarlint/core/telemetry/TelemetryHttpClient.java

```diff
@@ -39,10 +43,10 @@
 import org.sonarsource.sonarlint.core.telemetry.payload.cayc.CleanAsYouCodePayload;
 import org.sonarsource.sonarlint.core.telemetry.payload.cayc.NewCodeFocusPayload;
 
+@Named
+@Singleton
 public class TelemetryHttpClient {
```

Comment:
If we start using `SystemUtils` in `TelemetryHttpClient` (as suggested in another comment) we also can use it to detect OS in the `createPayload()` method to replace ` var os = System.getProperty("os.name");`

Reply:
After your comment, I noticed that `os` and `platform` values are the same in the telemetry payload. We can remove one of both. I'll create a ticket for it.

Reply:
Good point!